src/test/ui/did_you_mean/bad-assoc-ty.stderr

```diff
@@ -99,14 +99,24 @@ error[E0223]: ambiguous associated type
 LL | type G = dyn 'static + (Send)::AssocTy;
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use fully-qualified syntax: `<(dyn Send + 'static) as Trait>::AssocTy`
 
+warning: trait objects without an explicit `dyn` are deprecated
+  --> $DIR/bad-assoc-ty.rs:33:10
+   |
+LL | type H = Fn(u8) -> (u8)::Output;
+   |          ^^^^^^ help: use `dyn`: `<dyn Fn(u8)>`
```

Comment:
This one makes me uncomfortable, but I could not find how to fix it. Suggestions?

Reply:
This is outright incorrect, right? The whole type is `<dyn Fn(u8) -> (u8)::Output>`, if I'm reading this correctly. I think we can special case `Fn` traits in this format by looking at the `poly_trait_ref`.

It's also interesting that the error below is already making some suggestions, so it might make sense to only emit this one if E0233 isn't (but this might not be needed in this PR).